pkg/alert/ctl_alert_test.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright (C) 2019 Synopsys, Inc.
+
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements. See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership. The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied. See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+
+package alert
+
+import (
+	"fmt"
+
+	alertv1 "github.com/blackducksoftware/synopsys-operator/pkg/api/alert/v1"
+	crddefaults "github.com/blackducksoftware/synopsys-operator/pkg/util"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestNewAlertCtl(t *testing.T) {
+	assert := assert.New(t)
+	alertCtl := NewAlertCtl()
+	assert.Equal(&Ctl{
+		Spec:              &alertv1.AlertSpec{},
+		Registry:          "",
+		ImagePath:         "",
+		AlertImageName:    "",
+		AlertImageVersion: "",
+		CfsslImageName:    "",
+		CfsslImageVersion: "",
+		BlackduckHost:     "",
+		BlackduckUser:     "",
+		BlackduckPort:     0,
+		Port:              0,
+		StandAlone:        false,
+		AlertMemory:       "",
+		CfsslMemory:       "",
+		State:             "",
+	}, alertCtl)
+}
+
+func TestGetSpec(t *testing.T) {
+	assert := assert.New(t)
+	alertCtl := NewAlertCtl()
+	assert.Equal(alertv1.AlertSpec{}, alertCtl.GetSpec())
+}
+
+func TestSetSpec(t *testing.T) {
+	assert := assert.New(t)
+	alertCtl := NewAlertCtl()
+	specToSet := alertv1.AlertSpec{Namespace: "test", Registry: "test"}
+	alertCtl.SetSpec(specToSet)
+	assert.Equal(specToSet, alertCtl.GetSpec())
+
+	// check for error
+	assert.EqualError(alertCtl.SetSpec(""), "Error setting Alert Spec")
+}
+
+func TestCheckSpecFlags(t *testing.T) {
+	assert := assert.New(t)
+	alertCtl := NewAlertCtl()
+	specFlags := alertCtl.CheckSpecFlags()
+	assert.Nil(specFlags)
+}
+
+func TestSwitchSpec(t *testing.T) {
+	assert := assert.New(t)
+	alertCtl := NewAlertCtl()
+
+	var tests = []struct {
+		input    string
+		expected alertv1.AlertSpec
+	}{
+		{"empty", alertv1.AlertSpec{}},
```

Comment:
Better to explicitly define the assignments here for readability.  IE:

{
  input: "empty",
  expected : alertv1.AlertSpec{},
},

Reply:
Fixed :)